### PR TITLE
added checkstyle in project

### PR DESCRIPTION
**Pull request by ishitakeshawani**
added the checkstlye.xml
fixed the check style issues, there is no violation

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,6 +2,7 @@ plugins {
 	id 'org.springframework.boot' version '2.6.2'
 	id 'io.spring.dependency-management' version '1.0.11.RELEASE'
 	id 'java'
+	id	'checkstyle'
```
Comment (deepjoshi11th) [inline build.gradle]: remove additional spaces for consistency.

File: build.gradle [hunk 2]
```diff
@@ -10,11 +11,18 @@ sourceCompatibility = '11'
 
 repositories {
 	mavenCentral()
+	maven { url "https://plugins.gradle.org/m2/" }
```
Comment (deepjoshi11th) [inline build.gradle:14]: Thanks for adding more repo

File: build.gradle [hunk 3]
```diff
@@ -10,11 +11,18 @@ sourceCompatibility = '11'
...
+	maven { url "https://repo.spring.io/plugins-release/" }
 }
 
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+//	checkstyle 'package.to.my.check.module:mycheck-module:[version]'
```
Comment (deepjoshi11th) [inline build.gradle]: remove the comment

File: src/main/java/com/deep/sample/SampleApplication.java [hunk 4]
```diff
@@ -4,10 +4,16 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+@SuppressWarnings("checkstyle:")
```
Comment (deepjoshi11th) [inline src/main/java/com/deep/sample/SampleApplication.java]: Try to remove it.

File: src/main/java/com/deep/sample/model/Problem.java [hunk 5]
```diff
@@ -1,32 +1,60 @@
...
-import com.fasterxml.jackson.databind.annotation.JsonNaming;
 
 public class Problem {
     private String title;
     private String descriptionValue;
 
+    /**
+     * it is constructor here.
+     */
+
```
Comment (deepjoshi11th) [inline src/main/java/com/deep/sample/model/Problem.java]: don't have additional space between method and javadoc

File: src/main/java/com/deep/sample/service/SampleService.java [hunk 6]
```diff
@@ -14,6 +14,12 @@
         problemList.add(new Problem("Def", "Ghi"));
     }
 
+    /**
+     * it is getproblembyid.
+     * @param index
+     * @return it is returning value.
+     */
+
```
Comment (deepjoshi11th) [inline src/main/java/com/deep/sample/service/SampleService.java:22]: same as above

**Review by deepjoshi11th — COMMENTED** [review]
Given few minor comments.

File: src/test/java/com/deep/sample/service/SampleServiceTest.java [hunk 7]
```diff
@@ -5,13 +5,17 @@
...
     private SampleService sampleService;
+
+    /**
+     * it is testGetUser method.
+     */
     @Test
     public void testGetUser() {
         Problem problem = sampleService.getProblemById(0);
-        assert problem.getTitle()=="Abc";
+        assert problem.getTitle() == "Abc";
```
Comment (taanii) [inline src/test/java/com/deep/sample/service/SampleServiceTest.java:19]: Great work

File: src/main/java/com/deep/sample/model/Problem.java (hunk 5 above)
Comment (ishitakeshawani) [reply src/main/java/com/deep/sample/model/Problem.java]: completed

**Comment by ishitakeshawani** [comment]
Screenshot (61)

**Review by deepjoshi11th — APPROVED** [review]

**Pull request merged**